File: programmers/javascript/level3/네트워크.js
const n = 3;
const computers = [
  [1, 1, 0],
  [1, 1, 0],
  [0, 0, 1],
];

const dfs = (idx, arr, computers) => {
  //console.log(idx);
  arr[idx] = true;
  for (let i = 0; i < computers.length; i++) {
    if (computers[idx][i] === 1 && !arr[i]) {
      dfs(i, arr, computers);
    }
  }
};

const solution = (n, computers) => {
  let answer = 0;
  const arr = new Array(n).fill(false);
  //console.log(arr);

  while (arr.findIndex((e) => e === false) !== -1) {
    dfs(
      arr.findIndex((e) => e === false),
      arr,
      computers
    );
    answer++;
  }
  return answer;
};

console.log(solution(n, computers));

File: python/programmers/level3/네트워크.py
def solution(n, computers):
    def dfs(num):
        for i in range(n):
            if not vis[i] and computers[num][i] == 1:
                vis[i] = True
                dfs(i)

    answer = 0
    vis = [False] * n
    for i in range(n):
        if not vis[i]:
            vis[i] = True
            answer += 1
            dfs(i)
    return answer


print(solution(3, [[1, 1, 0], [1, 1, 0], [0, 0, 1]]))
